cmd/kops/create_cluster.go

```diff
@@ -444,6 +444,28 @@ func RunCreateCluster(f *util.Factory, out io.Writer, c *CreateClusterOptions) e
 			}
 			zoneToSubnetMap[zoneName] = subnet
 		}
+	} else if api.CloudProviderID(cluster.Spec.CloudProvider) == api.CloudProviderDO {
+		if len(c.Zones) > 1 {
+			return fmt.Errorf("digitalocean cloud provider currently only supports 1 region, expect multi-region support when digitalocean support is in beta")
+		}
+
+		// For DO we just pass in the region for --zones
+		region := c.Zones[0]
+		subnet := model.FindSubnet(cluster, region)
+
+		// for DO, subnets are just regions
+		subnetName := region
+
+		if subnet == nil {
+			subnet = &api.ClusterSubnetSpec{
+				Name: subnetName,
+				// region and zone are the same for DO
+				Region: region,
+				Zone:   region,
```

Comment:
Hmm... we should be able to cope without a Zone being set, but I'm guessing something breaks ?

Reply:
Yes, the [FindZonesForInstanceGroup](https://github.com/kubernetes/kops/blob/master/pkg/apis/kops/model/utils.go#L44) function only inserts to list if Zone is set, and so [this](https://github.com/kubernetes/kops/blob/master/pkg/model/master_volumes.go#L68) check fails. Any objections to add a region check there?

Reply:
No objections, particularly if we can scope it to DO.  The root issue there is that EBS volumes on AWS / PDs on GCE are single zone, and thus we need to figure out the zone.  But it looks like DO volumes are regional (wow!) so it's fine to relax that check.  I would want to see the existing checks for AWS & GCE continue to work (likely by a switch statement on the cloudproviderid).